src/cov/cov-report.c

```diff
@@ -1029,6 +1032,73 @@ static void cover_print_hierarchy_guts(FILE *f, cover_report_ctx_t *ctx)
               "   function GetExclude(excludeCmd) {\n"
               "      navigator.clipboard.writeText(excludeCmd);\n"
               "   }\n"
+              "   function sortTable(n, tableId, is_number) {\n"
```

Comment:
This code is quite ugly and bubble sort is the worst sorting algorithm. I wrote the following based on the SO answer here https://stackoverflow.com/a/49041392

```js
function sortTable(n, tableId) {
  const table = document.getElementById(tableId);
  const rows = Array.from(table.querySelectorAll('tr:nth-child(n+2)'));

  const getCellValue = (tr) => {
    const td = tr.getElementsByTagName('td')[n];
    return td.innerText || td.textContent;
  };

  rows.sort((a, b) => {
    const left = getCellValue(this.ascending ? a : b);
    const right = getCellValue(this.ascending ? b : a);

    if (left !== '' && right !== '' && !isNaN(left) && !isNaN(right)) {
      return left - right;
    } else {
      return left.localeCompare(right);
    }
  });

  rows.forEach(tr => table.appendChild(tr));

  this.ascending = !this.ascending;
}
```

Reply:
You are right, it is messy.

The alternative does not swap the integer percentages correctly. It treats `100 % (15/15)` as a string,
so it is the "lowest" first character. I had the `is_number` parameter for that purpose.

I updated the function to handle both string (sort by hierarchy name alphabetically) and int
(sort by box content split by `%` delimiter, sorts by number of percents reached).